### DISCUSSION TITLE
General discussion of NIST Parallel Panel results

### DISCUSSION THREAD

**Discussion by mcgratta** [MaCFP/macfp-db#417] [General]
@gmaragko82
@danmohad 
@TristanHehnen

Thanks for your submission of results for the Parallel Panel experiment. I am one of the session organizers and I am taking the lead on writing up a summary discussion of results and a presentation for the October workshop. 

There appear to be four submissions: NIST, FM, Ghent, and Julich. I assume that each of you have taken in the lead in generating and processing results. If not, let me know or add the appropriate person to conversation. I'd like to use this thread to discuss details concerning the summary discussion and presentation. 

First, a request. I would like each of us to plot our results in the same way. I used the four plot formats that were used by the experimentalists: [https://github.com/MaCFP/macfp-db/tree/master/Fire_Growth/NIST_Parallel_Panel/Documentation](https://github.com/MaCFP/macfp-db/tree/master/Fire_Growth/NIST_Parallel_Panel/Documentation)
You can find a python script to make each plot in my results folder: [https://github.com/MaCFP/macfp-db/tree/master/Fire_Growth/NIST_Parallel_Panel/Computational_Results/2023/NIST](https://github.com/MaCFP/macfp-db/tree/master/Fire_Growth/NIST_Parallel_Panel/Computational_Results/2023/NIST). You would need to copy my python script into your own folder and then replace the links to my data with links to your data in the script. I have already done this with the Ghent results, where I added a contour plot. 

I suggest that you run my python scripts to see what the end results look like, and then modify your own. We can fine-tune the plots (fonts, colors, etc) once we all have made them. I would like to focus our presentation of results on these four plots.

**Comment by danmohad** [comment]
Hi Kevin,

Thanks for taking the lead on this. It's definitely a good idea for our plots to be directly comparable. I have some comments and suggestions on these plots. 

- Mesh resolution and convergence
  - I think it's important to show the effect of mesh resolution on our results, and presenting experimental results along with a few levels of refinement is a powerful way to do this. However, this obviously means 2-3x the lines per plot, which quickly becomes very hard to read. `PMMA_heat_flux_profile.pdf` is an example of this. I think it would be useful to break that figure into three separate plots, grouped by HRR.
- HRR and Qrad plot
  - For similar reasons as above, I think HRR and Qrad should have their own plots. That way, the mesh resolution effect can be shown for both quantities without cluttering the plot. The NIST plot doesn't show Qrad from FDS right now; it would get very busy if it did.
- Burner heat flux profile
  - I'm inclined to think that 'height vs heat flux' this isn't the best way to visually represent the data. I don't think we gain much from seeing the shape of the profile; it will always be hyperbolic when plotted this way. Rather, I think what is more important is to be able to compare the distance from the experimental data points to the simulation data for different mesh resolutions, and to see how this comparison evolves over time, since this is indicative of the conjugate aspects of the problem, at different locations in space. Hence my preference to plot 'heat flux vs time' with different mesh resolutions, with separate plots for different heights. I'd like to hear others' thoughts on this.
- PMMA heat flux profile
  - Just to make sure I understand how this plot was generated: the HRR values used to choose the 'time' around which to extract the heat flux profile is only during the 'fire growth' stage, yes? I.e., only for {time s.t. HRR < max(HRR)}. And then you averaged for 2 seconds about each time {time s.t. HRR = HRR_i for HRR_i in [120kW, 200kW ...]}

**Comment by mcgratta** [comment]
I suggest we first just make the plots as I have and then we can work to improve them once we see how each of our results look. 

The HRR and Qrad plot can be thrown out. Let's just use the plotting script to make separate plots of HRR and Qrad.

**Comment by gmaragko82** [comment]
Hi Kevin, all,

Thanks for making the plot for UGent, much appreciated.

I agree that it's good to have the same plots so that the comparison is easier for people to follow. With this in mind, it might also be useful for the simulations that are presented to have the same baseline settings as in the experiments as well (e.g., burner HRR and duration the burner was on) when making the comparisons. This initial fire source seems to be rather important with respect to flame spread in this case. Any changes in the original setup can then perhaps be presented as additional results in order to show how sensitive the model predictions are. That's just my suggestion at least.

**Comment by mcgratta** [comment]
I agree. The flame spread in my simulations stopped when I turned off the burner as was done in the experiments. I plan to discuss this along with the grid dependence. Those two items seem to be the most important.

**Comment by mcgratta** [comment]
I need confirmation from each participant to make the four basic plots for this exercise, and when you expect to do it. I would then like to write the section in the summary document and assemble a presentation for the meeting. If we get those plots done quickly, we can have a good presentation and 90% of our section done by the time we meet in Japan.

**Reply by danmohad** [reply]
Confirmed. I'm working on it; should be done by EOD Monday.

**Comment by mcgratta** [comment]
Great. For the moment, you can just use the scripts in the NIST folder. Copy them to your folder and swap out our results for yours. It should change just a few lines. My python programming is not good enough to lump everyone's results into one big script.

**Comment by TristanHehnen** [comment]
Hi, I think I've got the four plots now. 
Unfortunately, the simulations are really slow so I'm not sure how much I can update the simulation results, thus the plots are a bit empty.

**Comment by mcgratta** [comment]
All part of the discussion. I think we'll all hit ourselves in the head at the meeting, realizing that we have overlooked something simple, and then return home and rerun. Hopefully at the next MaCFP, we will revisit this case plus all the others that we burned here at NIST. I prefer to look at several different fuels and configurations because it is always possible that there is something about this one fuel, PMMA, that we are missing.

**Comment by gmaragko82** [comment]
In general, I would say that this case has been quite challenging and difficult to replicate (at least for me). I will agree that we are either overlooking something 'simple', or there's something special about the fuel or there's a model deficiency of some sort (or something else). Perhaps a report from the experiments, together with additional data and tests, would help when revisiting the case for modelling in the future.

**Comment by mcgratta** [comment]
I started organizing the results. I need heat flux profiles from Gent. Also, Tristan, can you make your contour plot have the same shape and color scheme as the other three. You should be able to run everyone's python scripts to see them.

**Reply by gmaragko82** [reply]
You can go ahead with the limited data already provided from my side Kevin. My contribution for this case is just exploratory at this stage.

**Comment by TristanHehnen** [comment]
I've updated our simulation data and plots.

However, I strongly recommend to add a square `figsize` like [this](https://github.com/MaCFP/macfp-db/blob/f068194bf4fbed050d62c05d60d85bb7630cccb6/Fire_Growth/NIST_Parallel_Panel/Computational_Results/2023/BUWFZJ/plot_burner_heat_flux_contour.py#L86), to get a plot with the correct proportions instead of the default distorted plot.

**Comment by mcgratta** [comment]
Then redo all the contour plots. Use consistent colors, line types, and order of crude to fine resolution in the legend.

**Reply by TristanHehnen** [reply]
See #456

**Comment by TristanHehnen** [comment]
@mcgratta As discussed earlier, the contour plots and the HRR plots for the parallel panel setup are now unified. See PR #456 and PR #457 respectively.

**Comment by mcgratta** [comment]
I am putting together the presentation for the NIST Parallel Panel. Are all the latest results committed to the macfp-db repository?

**Reply by danmohad** [reply]
No changes here. I sent my poster to Randy last week, and just sent you a copy. It has some figures that didn't fit the scope of the PR, in case you want to use them too.

**Reply by gmaragko82** [reply]
No changes from my side either.

**Reply by TristanHehnen** [reply]
@mcgratta  See #470

**Comment by TristanHehnen** [comment]
Hi Kevin,
I've got an update on the panel simulation results, I'll submit it today in the afternoon.

**Comment by mcgratta** [comment]
I posted a copy of the Parallel Panel presentation [here](https://drive.google.com/drive/folders/1qy4_pXnaGPjesDwfXo8oT5RoUjw0yQxf). I don't know how to work on this presentation collaboratively without making a mess, so let me know what you want added, highlighted, and conclusions.

**Reply by danmohad** [reply]
Thanks for posting the presentation! Here are a few thoughts:

- I would make the plots on slide 11 bigger, they're hard to read right now
- A conclusion from slide 11 is that a mesh size of less than or equal to 1.3cm was needed to break down the 'pyramid-shaped' mean flame structure. Since Gent did not refine below 2.5cm, their results all have this shape.
- Another conclusion from slide 11 is that NIST's well-resolved predictions tend to decrease more rapidly away from the centerline than the experimental values, but on the centerline, their predictions are in very good agreement with experiment. Whereas FM's well-resolved predictions are much flatter laterally, in line with the experimental profiles, but underpredict the heat flux across heights.
- A conclusion from slide 12 is that FM's results for HRR are reasonably grid converged, while those of other groups are not
- Another conclusion from slide 12 is that NIST's prediction of the peak HRR value improved with refinement, with the most resolved prediction being very close to the experimental peak value, but the prediction of the timing of peak HRR got worse with refinement. FM's prediction of the peak HRR value and timing did not improve with refinement.

**Reply by TristanHehnen** [reply]
It may be worth noting that the NIST burner simulation with empty (Marinite) panels uses an oscillating boundary condition for the simulation domain. The BUWFZJ simulations use the default `OPEN` boundaries. The flatter contours in higher resolutions for BUWFZJ seem to be primarily the result of vortices that form near the panel edges. These vortices trap parts of the flames and pull them towards the edge, widening the profile. I've observed the vortices for fluid cells of 1 cm and less.

Also: the NIST simulations use the default simulation mode `VLES`, while the BUWFZJ uses `LES`. This also leads to different extinction models are being used which may have an impact due to varying oxygen concentrations.

I've started a discussion on the MaCFP google group a while ago. There I've also posted images of this:
https://groups.google.com/g/macfp-discussions/c/Hw6nyKv2YWo

Edit: typo

**Comment by mcgratta** [comment]
Thanks. I modified the slides and posted a new version.
